numpy/core/setup.py

```diff
@@ -605,6 +606,12 @@ def generate_api(ext, build_dir):
     config.add_include_dirs(join('src', 'umath'))
     config.add_include_dirs(join('src', 'npysort'))
 
+    # Need to add location of npy_math_internal.h, generated from src/npymath/npy_math_internal.h.src
+    npymath_build_dir = join('build', 'src.%s-%s.%s' % (sysconfig.get_platform(), sys.version_info[0], sys.version_info[1]),
```

Comment:
@juliantaylor Ok, good to know you can consider installing inlinable headers. So, probably this discussion is kind of misplaced since it does not relate to the code above for adding the include directory.

I don't insist we should work on installing inlinable headers right now and right here, I just want to clarify my understanding of your concerns. Let's assume we put inlinable headers into include/numpy/ directory. What I'm saying is that even if configuration is not inferred correctly in a 3rd party library, including npy_math.h with inlined implementation would not hurt because it has fallback implementations which can be enabled even in presence of compiler support when [some] built-ins are mistakenly not detected. Is your concern that these fall back implementations are somewhat less correct than built-ins which might be available but not used?
And there are certainly no problem when they are detected in some way correctly.

Should we could open an issue/feature request for documenting this discussion?